sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -521,21 +521,34 @@ private bool TryLoadConstructorArguments(AttributeData attributeData, IDictionar
                     {
                         case TypedConstantKind.Error:
                             break;
+
                         case TypedConstantKind.Primitive:
                             dict[argumentName] = arg.Value;
                             break;
+
                         case TypedConstantKind.Enum:
                             var enumValue = arg.Type!.GetMembers()
                               .FirstOrDefault(m => m is IFieldSymbol field
                                               && field.ConstantValue is object value
                                               && value.Equals(arg.Value));
+
+                            if (enumValue is null)
+                            {
+                                return false;
+                            }
+
+                            // we want just the enumValue symbol's name (Admin, Anonymous, Function)
```

Comment:
Just to make sure: this logic is not _specific_ to Http scenario? It will work for any enum?

Reply:
Yup not specific to the Http scenario